src/vm/olevariant.cpp

```diff
@@ -1301,7 +1304,7 @@ void OleVariant::MarshalBoolVariantOleToCom(VARIANT *pOleVariant,
 #endif // FEATURE_COMINTEROP
 
 void OleVariant::MarshalBoolArrayOleToCom(void *oleArray, BASEARRAYREF *pComArray,
-                                          MethodTable *pInterfaceMT)
+                                          MethodTable *pInterfaceMT, PCODE pManagedMarshalerCode)
```

Comment:
>pManagedMarshalerCode [](start = 75, length = 21)

It seems very few of these functions are using this parameter. What is the reason for them?

Reply:
These functions are passed up to the calling code through the `Marshaler` struct in `olevariant.h`. The design there is to have each field be a function pointer that points to each needed marshalling method. As a result, they all need to have the same signature, even if one of the parameters (such as this one) is unused. It's the same reason that every marshaler in olevariant.cpp has a `MethodTable*` parameter when only the interface marshaler uses it.

We can't resolve the struct marshalling stub from the `MethodTable*` in the marshaler since the lookup is slow enough to slow down non-blittable structure array marshalling by at least 600%.